sql/as_of_test.go

```diff
@@ -56,39 +58,50 @@ func TestAsOfTime(t *testing.T) {
 	}
 	tsEmpty := tm.Format(time.RFC3339Nano)
 	if _, err := db.Query(fmt.Sprintf(query, tsEmpty), 0); !testutils.IsError(err, `pq: database "d" does not exist`) {
-		t.Fatal("unexpected error:", err)
+		t.Fatal(err)
 	}
 
-	if _, err := db.Exec("CREATE DATABASE d"); err != nil {
-		t.Fatal("unexpected error:", err)
-	}
-	if err := db.QueryRow("SELECT now()").Scan(&tm); err != nil {
+	var logicalTS int64
+	if err := db.QueryRow("CREATE DATABASE d; SELECT cluster_logical_timestamp()::int").Scan(&logicalTS); err != nil {
 		t.Fatal(err)
 	}
-	tsDBExists := tm.Format(time.RFC3339Nano)
+	// Make sure we see the DB create by using a TS after its txn time.
+	tsDBExists := time.Unix(0, logicalTS+1).Format(time.RFC3339Nano)
+	// Wait until now() is after the above cluster_logical_timestamp() so the max check doesn't get hit.
+	util.SucceedsSoon(t, func() error {
```

Comment:
We just finished a 3 revision process adding and removing this as a function. See Tamir's comments above about error line numbers.